### ISSUE TITLE
Passing options down the pipe

### ISSUE THREAD

**Issue by clibu** [bevry/caterpillar#126]
It would be useful to be able to pass options along the pipe. 

For example ``Human`` has an option ``color`` as does ``Browser``, which would typically be set the same. Both default to ``true``, however if you want them to be ``false`` then you need to include ``{ color: false }`` for both.

Instead if ``Human( opts )`` could be passed to ``Browser`` etc. both client code and usage would be simplified as  ``{ color: false }`` would only need to be specified once.

Similarly it would be handy if ``LogEntry`` could be passed down the pipe then the ``Browser`` transform could access ``LogEntry`` passed to ``Human()``.

Finally it would be useful if Transform's could access Logger options.

I do however realize that these suggestions maybe outside the design scope or even unwanted.

**Comment by balupton** [comment]
Prior to version 6 we had this, however it was removed as it was too difficult to reason about and maintain. Much clearer for options to not be inherited and merged, or at least leaving such to userland instead of core where they know the assumptions about merging and inheritance.

**Comment by balupton** [comment]
The browser transform and the hunan transform could certainly be merged. The plan would be for an ansi package that outputs the style for the browser if in a browser environment. And either the one package does it automatically, or you configure the new human transform with the appropriate config for the environment.

**Comment by clibu** [comment]
Yes, I'm using [ansicolor](https://github.com/xpl/ansicolor) which replaces your ansi code. Works a treat. 

I now have a combined human and browser module called _superHuman_ 😀 which has target _(browser/node)_ options amongst others.  It's coming along nicely. 

It can either do the original colored and expanded output or the user expandable compact output like this:

![image](https://user-images.githubusercontent.com/12797550/96327500-fe6a3480-1085-11eb-904e-782ece7c6f98.png)

**Comment by balupton** [comment]
Lookin good
